### PR TITLE
Add securityContexts to secureCodeBox Components

### PR DESCRIPTION
**Pull request by J12934**
Add configurable [Kubernetes securityContext](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to all included pods.

For all direct "secureCodeBox" components (Operator, Lurcher, Parser & Hooks) these are completely pre-configured to enable all securityContext features which are generally available on all supported Kubernetes Versions.

For the scanner container these have not yet been configured, as this might required more tweaking to properly support the individual scanner without messing with their functionality. For now only Nmap has been configured with a proper securityContext. As the securityContext is exposed via the Helm values it can be configured individually by the user.

### CODE REVIEW

**Comment by J12934** [comment]
While adding the securityContext i've noticed that all scanners had `env`, `extraVolumes`, `extraVolumeMounts`, `extraContainers` config values in their Helm values but their templates didn't really support them. I've added them to all integrated scanners.

File: operator/README.md [hunk 1]
```diff
@@ -38,10 +38,10 @@ helm install securecodebox-operator secureCodeBox/operator
 | s3.port | string | `nil` |  |
 | s3.secretAttributeNames.accesskey | string | `"accesskey"` |  |
 | s3.secretAttributeNames.secretkey | string | `"secretkey"` |  |
-| securityContext.allowPrivilegeEscalation | bool | `false` |  |
-| securityContext.capabilities.drop[0] | string | `"all"` |  |
-| securityContext.privileged | bool | `false` |  |
-| securityContext.readOnlyRootFilesystem | bool | `true` |  |
-| securityContext.runAsNonRoot | bool | `true` |  |
+| securityContext.allowPrivilegeEscalation | bool | `false` | Ensures that users privilidges canout be escalated |
```
Comment (nigthknight) [inline operator/README.md]: Ensure that users privileges cannot be escalated

File: operator/README.md [hunk 2]
```diff
@@ -38,10 +38,10 @@ helm install securecodebox-operator secureCodeBox/operator
...
+| securityContext.capabilities.drop[0] | string | `"all"` | This drops all linux privilidges from the operator container. They are not required |
```
Comment (nigthknight) [inline operator/README.md]: privileges

File: operator/README.md [hunk 3]
```diff
@@ -38,10 +38,10 @@ helm install securecodebox-operator secureCodeBox/operator
...
+| securityContext.privileged | bool | `false` | Ensures that the operator container is not run in privilidged mode |
```
Comment (nigthknight) [inline operator/README.md]: privileged

File: scanners/nmap/README.md [hunk 4]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
 
+## Operating System Scans
+
+:::caution
+Warning! This is currently not tested and might require additional testing to work 😕
+:::
+
+If you want to use Nmap to identify operating systems of hosts you'll need to weaken the securityContext config, as Nmap requires the capability to send raw sockets to identify operating systems. See [Nmap Docs](https://secwiki.org/w/Running_nmap_as_an_unprivileged_user)
+
+You can deployed the ScanType with the config like this:
```
Comment (nigthknight) [inline scanners/nmap/README.md]: You can deploy...

File: scanners/nmap/README.md [hunk 5]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
+
+```bash
+cat <<EOF | helm install nmap-privilidged ./scanners/nmap --values -
```
Comment (nigthknight) [inline scanners/nmap/README.md]: nmap-privileged

File: scanners/nmap/README.md [hunk 6]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
+
+You the start scans with operating system identification enabled:
+
+```yaml
+apiVersion: "execution.experimental.securecodebox.io/v1"
+kind: Scan
+metadata:
+  name: "nmap-os-scan"
+spec:
+  scanType: "nmap-privilidged"
```
Comment (nigthknight) [inline scanners/nmap/README.md]: nmap-priviledged

File: scanners/nmap/README.md [hunk 7]
```diff
@@ -53,4 +95,9 @@ Some useful example parameters listed below:
 | scannerJob.extraVolumeMounts | list | `[]` | Optional VolumeMounts mapped into each scanJob (see: https://kubernetes.io/docs/concepts/storage/volumes/) |
 | scannerJob.extraVolumes | list | `[]` | Optional Volumes mapped into each scanJob (see: https://kubernetes.io/docs/concepts/storage/volumes/) |
 | scannerJob.resources | object | `{}` | CPU/memory resource requests/limits (see: https://kubernetes.io/docs/tasks/configure-pod-container/assign-memory-resource/, https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/) |
+| scannerJob.securityContext.allowPrivilegeEscalation | bool | `false` | Ensures that users privilidges canout be escalated |
```
Comment (nigthknight) [inline scanners/nmap/README.md]: privileged

File: scanners/nmap/README.md [hunk 8]
```diff
@@ -53,4 +95,9 @@ Some useful example parameters listed below:
...
+| scannerJob.securityContext.capabilities.drop[0] | string | `"all"` | This drops all linux privilidges from the container. |
```
Comment (nigthknight) [inline scanners/nmap/README.md]: privileged

File: scanners/nmap/README.md [hunk 9]
```diff
@@ -53,4 +95,9 @@ Some useful example parameters listed below:
...
+| scannerJob.securityContext.privileged | bool | `false` | Ensures that the scanner container is not run in privilidged mode |
```
Comment (nigthknight) [inline scanners/nmap/README.md]: privileged

File: hook-sdk/nodejs/Dockerfile [hunk 10]
```diff
@@ -5,10 +5,10 @@ RUN npm ci --production
...
 ARG NODE_ENV
-RUN addgroup -S app && adduser app -S -G app
+RUN addgroup --system --gid 1001 app && adduser app --system --uid 1001 --ingroup app
 WORKDIR /home/app/hook-wrapper/
 COPY --from=build --chown=app:app /home/app/node_modules/ ./node_modules/
 COPY --chown=app:app ./hook-wrapper.js ./hook-wrapper.js
-USER app
+USER 1001
 ENV NODE_ENV ${NODE_ENV:-production}
 ENTRYPOINT ["node", "/home/app/hook-wrapper/hook-wrapper.js"]
```
Comment (nigthknight) [inline hook-sdk/nodejs/Dockerfile]: Newline at EOF missing

File: operator/README.md [hunk 11]
```diff
@@ -21,7 +21,7 @@ helm install securecodebox-operator secureCodeBox/operator
 
 | Key | Type | Default | Description |
 |-----|------|---------|-------------|
-| image.pullPolicy | string | `"Always"` | Image pull policy |
```
Comment (nigthknight) [inline operator/README.md:24]: Maybe revert to Always

File: operator/values.yaml [hunk 12]
```diff
@@ -12,7 +12,21 @@ image:
   # @default -- defaults to the charts version
   tag: null
   # image.pullPolicy -- Image pull policy
-  pullPolicy: Always
```
Comment (nigthknight) [inline operator/values.yaml:15]: Maybe switch back to Always

File: parser-sdk/nodejs/Dockerfile [hunk 13]
```diff
@@ -5,10 +5,10 @@ RUN npm ci --production
...
 ARG NODE_ENV
-RUN addgroup -S app && adduser app -S -G app
+RUN addgroup --system --gid 1001 app && adduser app --system --uid 1001 --ingroup app
 WORKDIR /home/app/parser-wrapper/
 COPY --from=build --chown=app:app /home/app/node_modules/ ./node_modules/
 COPY --chown=app:app ./parser-wrapper.js ./parser-wrapper.js
-USER app
+USER 1001
 ENV NODE_ENV ${NODE_ENV:-production}
 ENTRYPOINT ["node", "/home/app/parser-wrapper/parser-wrapper.js"]
```
Comment (nigthknight) [inline parser-sdk/nodejs/Dockerfile]: Newline at EOF missing

File: scanners/nmap/README.md.gotmpl [hunk 14]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
 
+## Operating System Scans
+
+:::caution
+Warning! This is currently not tested and might require additional testing to work 😕
+:::
+
+If you want to use Nmap to identify operating systems of hosts you'll need to weaken the securityContext config, as Nmap requires the capability to send raw sockets to identify operating systems. See [Nmap Docs](https://secwiki.org/w/Running_nmap_as_an_unprivileged_user)
+
+You can deployed the ScanType with the config like this:
```
Comment (nigthknight) [inline scanners/nmap/README.md.gotmpl]: You can deploy [..]

File: scanners/nmap/README.md.gotmpl [hunk 15]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
+
+```bash
+cat <<EOF | helm install nmap-privilidged ./scanners/nmap --values -
```
Comment (nigthknight) [inline scanners/nmap/README.md.gotmpl]: nmap-privileged

File: scanners/nmap/README.md.gotmpl [hunk 16]
```diff
@@ -42,6 +42,48 @@ Some useful example parameters listed below:
...
+  scanType: "nmap-privilidged"
+  parameters:
+    - --privileged
+    - "-O"
+    - www.iteratec.de
+```
+
 ## Chart Configuration
 
 {{ template "chart.valuesTable" . }}
```
Comment (nigthknight) [inline scanners/nmap/README.md.gotmpl]: Newline at EOF missing

File: scanners/nmap/scanner/Dockerfile [hunk 17]
```diff
@@ -1,5 +1,5 @@
 FROM alpine:3.12
 RUN apk add --no-cache nmap=7.80-r2 nmap-scripts=7.80-r2
-RUN addgroup -S nmap && adduser -S -g nmap nmap
-USER nmap
+RUN addgroup --system --gid 1001 nmap && adduser nmap --system --uid 1001 --ingroup nmap
+USER 1001
 CMD [nmap]
```
Comment (nigthknight) [inline scanners/nmap/scanner/Dockerfile]: Newline at EOF missing

File: scanners/sslyze/templates/sslyze-scan-type.yaml [hunk 18]
```diff
@@ -23,3 +23,14 @@ spec:
...
+                {{- toYaml .Values.scannerJob.securityContext | nindent 16 }}
+              env:
+                {{- toYaml .Values.scannerJob.env | nindent 16 }}
+              volumeMounts:
+                {{- toYaml .Values.scannerJob.extraVolumeMounts | nindent 16 }}
+            {{- if .Values.scannerJob.extraContainers }}
+            {{- toYaml .Values.scannerJob.extraContainers | nindent 12 }}
+            {{- end }}
+          volumes:
+            {{- toYaml .Values.scannerJob.extraVolumes | nindent 12 }}
```
Comment (nigthknight) [inline scanners/sslyze/templates/sslyze-scan-type.yaml]: Newline at EOF missing

File: scanners/test-scan/templates/test-scan-scan-type.yaml [hunk 19]
```diff
@@ -21,3 +21,14 @@ spec:
...
+                {{- toYaml .Values.scannerJob.securityContext | nindent 16 }}
+              env:
+                {{- toYaml .Values.scannerJob.env | nindent 16 }}
+              volumeMounts:
+                {{- toYaml .Values.scannerJob.extraVolumeMounts | nindent 16 }}
+            {{- if .Values.scannerJob.extraContainers }}
+            {{- toYaml .Values.scannerJob.extraContainers | nindent 12 }}
+            {{- end }}
+          volumes:
+            {{- toYaml .Values.scannerJob.extraVolumes | nindent 12 }}
```
Comment (nigthknight) [inline scanners/test-scan/templates/test-scan-scan-type.yaml]: Newline at EOF missing

File: scanners/trivy/templates/trivy-scan-type.yaml [hunk 20]
```diff
@@ -28,3 +28,14 @@ spec:
...
+                {{- toYaml .Values.scannerJob.securityContext | nindent 16 }}
+              env:
+                {{- toYaml .Values.scannerJob.env | nindent 16 }}
+              volumeMounts:
+                {{- toYaml .Values.scannerJob.extraVolumeMounts | nindent 16 }}
+            {{- if .Values.scannerJob.extraContainers }}
+            {{- toYaml .Values.scannerJob.extraContainers | nindent 12 }}
+            {{- end }}
+          volumes:
+            {{- toYaml .Values.scannerJob.extraVolumes | nindent 12 }}
```
Comment (nigthknight) [inline scanners/trivy/templates/trivy-scan-type.yaml]: Newline at EOF missing

File: scanners/wpscan/templates/wpscan-scan-type.yaml [hunk 21]
```diff
@@ -25,4 +25,15 @@ spec:
...
+                {{- toYaml .Values.scannerJob.securityContext | nindent 16 }}
+              env:
+                {{- toYaml .Values.scannerJob.env | nindent 16 }}
+              volumeMounts:
+                {{- toYaml .Values.scannerJob.extraVolumeMounts | nindent 16 }}
+            {{- if .Values.scannerJob.extraContainers }}
+            {{- toYaml .Values.scannerJob.extraContainers | nindent 12 }}
+            {{- end }}
+          volumes:
+            {{- toYaml .Values.scannerJob.extraVolumes | nindent 12 }}
```
Comment (nigthknight) [inline scanners/wpscan/templates/wpscan-scan-type.yaml]: Newline at EOF missing

**Review by nigthknight — CHANGES_REQUESTED** [review]

**Review by nigthknight — APPROVED** [review]

**Pull request merged**